driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -36,14 +37,29 @@ impl LinearOptimisationPriceFinder {
     }
 }
 
-fn token_id(token: usize) -> String {
+fn token_id(token: u16) -> String {
     format!("token{}", token)
 }
 
 fn account_id(account: H160) -> String {
     format!("{:x}", account)
 }
 
+fn serialize_tokens(orders: &[models::Order]) -> Vec<String> {
+    // Get collection of all token ids appearing in orders
+    let mut token_ids = orders.iter().map(|o| o.buy_token).collect::<Vec<u16>>();
+    token_ids.extend(orders.iter().map(|o| o.sell_token).collect::<Vec<u16>>());
+
+    // Remove duplicate tokens by casting to HashSet and convert back to Vec
+    let unique_token_ids: HashSet<u16> = HashSet::from_iter(token_ids.iter().cloned());
+    let mut token_vec = unique_token_ids.into_iter().collect::<Vec<u16>>();
+
+    // unstable sort has no performance loss since elements are unique
+    token_vec.sort_unstable();
```

Comment:
Is this needed? Afaik the solver doesn't care about the order.

Reply:
Yes, agreed!